### DISCUSSION TITLE
Is there a way to stop pocketbase from accepting requests for a while?

### DISCUSSION THREAD

**Discussion by fbede** [pocketbase/pocketbase#4837] [Q&A] [answered]
I'm gradually warming up to running pb in production so I was thinking, what if I want to carry out some maintenance on pocketbase, is their a way to temporarily block requests that are coming from the API? The reason I am asking is to prevent users from accessing the server while I might be making changes to the collection schema for example...

**Comment by ganigeorgiev** [comment, marked as answer]
> is their a way to temporarily block requests that are coming from the API?

There is no such option at the moment (_there are plans to repurpose the "Hide controls ..." settings toggle to "Production mode" and I will consider adding a "Maintanence mode" with it_).

> The reason I am asking is to prevent users from accessing the server while I might be making changes to the collection schema for example

While direct collection changes on production are generally safe, the recommended way to deploy schema changes are through migrations ([Go](https://pocketbase.io/docs/go-migrations/) or [JS](https://pocketbase.io/docs/js-migrations/)). This ensures that the collections in your dev and prod environment stays in sync.
